### ISSUE TITLE
Bermuda Stops recieving Advertisments from BT Proxys after a few minutes

### ISSUE THREAD

**Issue by Kroeger02** [agittins/bermuda#403]
## Configuration

![image](https://github.com/user-attachments/assets/b75ad0a9-2cd5-41aa-91a3-bd448338d9be)
![image](https://github.com/user-attachments/assets/d04f7d24-190a-43fc-a5df-31264c16212b)

## Describe the bug

I use 2 ESP32 as BT Proxys, after a short time Bermuda stops recieving Advertisments from the Proxy. After a reboot of the ESPs it works for a few minutes.

## Diagnostics

I Installed the ESPs via the ESP Bluetooth Installer Website 

https://esphome.io/projects/?type=bluetooth

After that i updated them to the latest Firmware via ESPHome in HA

See attached File

[config_entry-bermuda-01JDYQTPGVRWN70DE9W5799XW2.json](https://github.com/user-attachments/files/17965881/config_entry-bermuda-01JDYQTPGVRWN70DE9W5799XW2.json)

**Comment by agittins** [comment]
Hi, sorry for the delay in following up with you.

The issue as you have already worked out seems to be with the esp32's, as Bermuda is seeing them fine to start with, and also still sees the local bluetooth adaptor ok.

I think the problem will be localised at the esp32 devices themselves.

The esphome ready-made projects thingie is great, but the settings aren't well-tuned for Bermuda.

When you first installed esphome onto the ESP32's, did you use an OTA update over wifi, or did you use a USB cable to serial-flash them? It's important to flash them at least once using a serial cable, as there are some flash formatting changes that only happen during a serial update (that only needs to happen once) but they can't be done via OTA updates.

Are they definitely standard ESP32 boards (not S3 or C3 variants)? If so, in your esp32 yaml, if you include this section:

```
packages:
  Bermuda.esp32: github://agittins/bermuda-proxies/packages/bermuda-proxy-esp32-orig-minimal.yaml
```

(you can replace `orig` wtih `s3` or `c3` if those are the boards you have) and re-flash - ideally using a USB connection.

If you have any trouble doing that, or if it doesn't solve it, please post your complete esphome yaml file and we'll see what we can work out.

**Comment by Kroeger02** [comment]
Thanks for your reply. And its no problem that the answer tooks a little bit longer, i`m happy about getting an answer :)

If i take a look to the data sheet of my ESP i have an ESP32 (not S3 or C3).

My ESP32s are the following ones:

diymore ESP32 Mini Development Board ESP32 NodeMCU WiFi Bluetooth ESP32 CH9102F Chip Typ C

The first time commissioning and flashing was via USB and the esp ready-made Project. After that i did an firmwareupdate via OTA to all of the ESPs.

Your named content of the section i couldn`T find in my yamls.

My have the following Content.


Proxy 1
`substitutions:
  name: esp32-bluetooth-proxy-2c6b5c
  friendly_name: BT_Proxy_Wohnzimmer_ESP32
packages:
  esphome.bluetooth-proxy: github://esphome/bluetooth-proxies/esp32-generic/esp32-generic.yaml@main
esphome:
  name: ${name}
  name_add_mac_suffix: false
  friendly_name: ${friendly_name}
api:
  encryption:
    key: (removed for security reasons)


wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password`
  
  Proxy 2
`substitutions:
  name: esp32-bluetooth-proxy-2d1d3c
  friendly_name: BT_Proxy_Arbeitszimmer_ESP32
packages:
  esphome.bluetooth-proxy: github://esphome/bluetooth-proxies/esp32-generic/esp32-generic.yaml@main
esphome:
  name: ${name}
  name_add_mac_suffix: false
  friendly_name: ${friendly_name}

api:
  encryption:
    key: (removed for security reasons)

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password`
  
  Proxy 3
  `substitutions:
  name: esp32-bluetooth-proxy-2d4578
  friendly_name: BT_Proxy_Flur_ESP32
packages:
  esphome.bluetooth-proxy: github://esphome/bluetooth-proxies/esp32-generic/esp32-generic.yaml@main
esphome:
  name: ${name}
  name_add_mac_suffix: false
  friendly_name: ${friendly_name}
api:
  encryption:
    key: (removed for security reasons)

logger:
  level: DEBUG

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password`
  
  If i should flash them again via USB, i need to take a look how to do this :)

**Comment by agittins** [comment]
No worries. Those configs look good to me. I'd try flashing that via OTA, and see how it goes.

For a USB update, when you click "INSTALL" does it present you with a list of options like this?
![image](https://github.com/user-attachments/assets/9f59bd38-9ece-4f50-af86-5da808c55dfe)

The one with the arrow is probably what you want to choose for a usb serial install. The language is a bit confusing!

If you are seeing a different interface from that maybe include some screenshots and I'll see if I can work out the steps for you.

**Comment by Kroeger02** [comment]
Hey,

thanks for your advice. I can see this window in my Home Assistant ESP Home Dashboard, should i choose this?

Or better what is the next step?

**Comment by agittins** [comment]
Yep, just choose that option. The process should then be very much like when you first flashed the board - you plug it in via usb, and it will ask you to select the device from a list of serial devices. If you unplug/plug the board at that point you should be able to tell which serial port it is.

**Comment by Kroeger02** [comment]
I clicked on "Install" on ESP Home. But i think i didn`t solve the Problem. The error still occours.

![image](https://github.com/user-attachments/assets/50b5d07d-bd13-4b20-8b38-3a7c6b2e5829)
![image](https://github.com/user-attachments/assets/1d9c6c11-8482-45d3-b2b3-91a31bb45f17)
![image](https://github.com/user-attachments/assets/05640c67-d21e-4205-8f7b-ff6b6a8eef74)

Or do you think i should install a special bermuda Package?

**Comment by Kroeger02** [comment]
Someone has further Ideas?

Or have i done something wrong?

**Comment by agittins** [comment]
Hi, sorry for the delay in getting back to you, the whole christmas/new years thing has been crazy!

Taking another look at your config files, I'd suggest removing the existing `packages` section and replacing it with this:

```yaml
packages:
  Bermuda.esp32: github://agittins/bermuda-proxies/packages/bermuda-proxy-esp32-orig-minimal.yaml
```

This does a similar thing, but has improvements specifically to make things perform better with Bermuda.

In your previous post, your last screenshot showed a thing about "Install your existing ESPHome project" - I'm not familiar with that dialog, is that what it showed after you chose "Plug into this computer" or was it from a different process?

You might find that goes away if you use the `packages` section above.

**Comment by agittins** [comment]
That's great to hear you got it sorted! I'm not sure why the two integrations would interact in that way, but good to know it's something to keep an eye out for. I'll close this issue out since your problem seems to be resolved.
